### ISSUE TITLE
Packaging static library causes "unable to run linkage report for this type of binary" error on Windows

### ISSUE THREAD

**Issue by penberg** [axodotdev/cargo-dist#1356]
I have the following `cargo dist` config:

```
# Config for 'cargo dist'
[workspace.metadata.dist]
# The preferred cargo-dist version to use in CI (Cargo.toml SemVer syntax)
cargo-dist-version = "0.21.0"
# CI backends to support
ci = "github"
# Which kinds of built libraries to include in the final archives
package-libraries = "cstaticlib"
# Which kinds of packaged libraries to install
install-libraries = "cstaticlib"
# The installers to generate for each app
installers = ["shell", "powershell"]
# Target platforms to build apps for (Rust target-triple syntax)
targets = ["aarch64-apple-darwin", "x86_64-apple-darwin", "x86_64-unknown-linux-gnu", "x86_64-pc-windows-msvc"]
# Which actions to run on pull requests
pr-run-mode = "plan"
# Path that installers should place binaries in
install-path = "~/.limbo"
# Whether to install an updater program
install-updater = true
# Whether to consider the binaries in a package for distribution (defaults true)
dist = false
# Whether to enable GitHub Attestations
github-attestations = true
```

But it's failing on Windows:

```
warning: `limbo-wasm` (lib) generated 1 warning
   Compiling limbo_sim v0.0.4 (D:\a\limbo\limbo\simulator)
warning: `py-limbo` (lib) generated 4 warnings
    Finished `dist` profile [optimized] target(s) in 3m 50s
  × unable to run linkage report for this type of binary

cargo dist ran successfully
Error: Process completed with exit code 1.
```

Full log here:

https://github.com/penberg/limbo/actions/runs/10510564356/job/29119512466

**Comment by penberg** [comment]
@Gankra Heh, I am happy to debug this more if needed, just let me know. I was in a rush to get a release out so had to quickly fill out a bug report and disable the static library packaging for now.

**Comment by Gankra** [comment]
The issue is rooted in us returning an error here: 

https://github.com/axodotdev/cargo-dist/blob/921905e7b8e9927a8cbbea25e9f0f2847e9410fd/cargo-dist/src/linkage.rs#L370-L376

To make linkage *work* someone needs to investigate why we're not getting what we want (is there a variant we should be handling?).

To make it non-fatal there's someone higher in the callstack that should be treating the error as non-fatal. I'd need to refamiliarize myself with this code to find the exact place where we tend to go "it's ok if this fails, move along" in the linkage subsystem.

**Comment by Gankra** [comment]
tbc: making it non-fatal is the more important thing here.

**Comment by mistydemeo** [comment]
I see what the issue is here, on both angles. I'll be putting up a branch soon which a) makes failing to gather linkage nonfatal, and b) handles static libraries properly on Windows.

**Comment by mistydemeo** [comment]
I've released a fix for this in a fresh prerelease, 0.22.0-prerelease.1. I've also checked locally and confirmed that this works now. Sorry for the inconvenience!
